SonarQube.Common/RoslynV1SarifFixer.cs

```diff
@@ -0,0 +1,107 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SarifUtilities.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Diagnostics;
+
+namespace SonarQube.Common
+{
+    public static class RoslynV1SarifFixer
+    {
+        /// <summary>
+        /// Returns true if the given SARIF came from the VS 2015 RTM Roslyn, which does not provide correct output.
+        /// </summary>
+        public /* for test */ static bool IsSarifFromRoslynV1(string input)
+        {
+            // low risk of false positives / false negatives
+            return (input.Contains(@"""toolName"": ""Microsoft (R) Visual C# Compiler""")
+                && input.Contains(@"""productVersion"": ""1.0.0"""));
+        }
+
+        /// <summary>
+        /// Used to correct invalid SARIF generated by VS 2015 RTM Roslyn. 
+        /// Correctly escapes backslashes and quotes.
+        /// 
+        /// Returns true if the output is valid SARIF (even if there was no change).
+        /// 
+        /// Example invalid line:
+        /// "shortMessage": "message \test\ ["_"]",
+        /// 
+        /// Example valid line:
+        /// "shortMessage": "message \\test\\ [\"_\"]",
+        /// </summary>
+        public /* for test */ static bool FixRoslynV1Sarif(string input, out string output)
+        {
+            try {
+                JObject.Parse(input);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Correct the spelling of 'Sarif' in member name 'RoslynV1SarifFixer.FixRoslynV1Sarif(string, out string)' or remove it entirely if it represents any sort of Hungarian notation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AIdentifiersShouldBeSpelledCorrectly)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Consider a design that does not require that 'output' be an out parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AAvoidOutParameters)